### PR TITLE
Add benchmarks for Watch

### PR DESCRIPTION
**Pull request by DanG100**
* Add test for Watch and WatchAll

Note: checks intentional don't run every loop so that its not including the timing of the benchmark

### CODE REVIEW

File: ygnmi/bench_test.go [hunk 1]
```diff
@@ -28,24 +29,31 @@ import (
...
 
+// subClient will send count notifications, then return EOF
+// if count > len(resp), the last notification is sent repeatedly
 type subClient struct {
 	gpb.GNMI_SubscribeClient
-	resp *gpb.SubscribeResponse
-	sent bool
+	resp  []*gpb.SubscribeResponse
+	count int
+	i     int
```
Comment (greg-dennis) [inline ygnmi/bench_test.go:38]: the name "i" is a little opaque. maybe "recvs"?

File: ygnmi/bench_test.go [hunk 2]
```diff
@@ -74,34 +82,124 @@ func BenchmarkGet(b *testing.B) {
...
+		ctx := context.Background()
+		q := root.New().Model().SingleKeyAny().State()
+		setUpdate(testutil.GNMIPath(b, "/model/a/single-key[key=\"foo\"]/state/value"), &gpb.TypedValue{Value: &gpb.TypedValue_IntVal{IntVal: 1}})
+		var got []*exampleoc.Model_SingleKey
+		for i := 0; i < b.N; i++ {
+			got, _ = ygnmi.GetAll(ctx, c, q)
+		}
+		b.StopTimer()
+		if *got[0].Value != 1 {
+			b.Fatalf("got %v want %v", *got[0].Value, 1)
```
Comment (greg-dennis) [inline ygnmi/bench_test.go:124]: "GetAll unexpected result: got %v ...

File: ygnmi/bench_test.go [hunk 3]
```diff
@@ -74,34 +82,124 @@ func BenchmarkGet(b *testing.B) {
...
-			if *got[0].Value != 1 {
-				b.Fatalf("got %v want %v", *got[0].Value, 1)
-			}
+			got, _ = ygnmi.WatchAll(ctx, c, q, func(v *ygnmi.Value[*exampleoc.Model_SingleKey]) error {
+				return ygnmi.Continue
+			}).Await()
+		}
+		b.StopTimer()
+		if v, _ := got.Val(); *v.Value != 1 {
+			b.Fatalf("Watch unexpected result: got %v want %v", v.Value, 1)
```
Comment (greg-dennis) [inline ygnmi/bench_test.go:202]: *WatchAll

**Review by greg-dennis — APPROVED** [review]

File: ygnmi/bench_test.go [hunk 4]
```diff
@@ -28,24 +29,31 @@ import (
 	gpb "github.com/openconfig/gnmi/proto/gnmi"
 )
 
+// subClient will send count notifications, then return EOF
+// if count > len(resp), the last notification is sent repeatedly
```
Comment (wenovus) [inline ygnmi/bench_test.go:33]: could you add a period after the first sentence? I thought it was a full sentence and was wondering why it returns EOF if count > len(resp)

File: ygnmi/bench_test.go [hunk 5]
```diff
@@ -28,24 +29,31 @@ import (
...
 type subClient struct {
 	gpb.GNMI_SubscribeClient
-	resp *gpb.SubscribeResponse
-	sent bool
+	resp  []*gpb.SubscribeResponse
+	count int
```
Comment (wenovus) [inline ygnmi/bench_test.go:37]: It's not immediately obvious to me what this means. I'm guessing something like "the number of non-EOF responses to be received by the client"?

**Review by wenovus — APPROVED** [review]

File: ygnmi/bench_test.go (hunk 5 above)
Comment (wenovus) [reply ygnmi/bench_test.go:37]: I meant the "count" field.

Comment (DanG100) [reply ygnmi/bench_test.go]: renamed count to numResponses and added comment

**Pull request merged**